### ISSUE TITLE
Steamwebhelper crash - no gui - libdrm.so.2

### ISSUE THREAD

**Issue by longshanks197** [ValveSoftware/steam-for-linux#10569]
#### Your system information

* Steam client version (build number or date): 1709168962
* Distribution (e.g. Ubuntu): Devuan 5 (based on Debian Bookworm)
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes
* Steam Logs: [steam-logs.tar.gz](https://github.com/ValveSoftware/steam-for-linux/files/14470655/steam-logs.tar.gz)
* GPU: AMD Radeon RX 7800 XT/ Ryzen 7700

#### Please describe your issue in as much detail as possible:
Steam failed to launch a few days ago when the latest stable build dropped. There was no change to my directory structure. With the latest responses and closing due to a symbolic links, this does not apply to my setup. In my home folder ".steam" is mounted via fstab to a dedicated drive.

AMD drivers are direct from AMD at version 23.40.2/6.0.60002. Drivers rely on /opt folder which may be the issue.

Launching Steam generates the steamwebhelper is not responding dialog. Options presented does not change result. 

Steam does log in and tray icon appears but cannot launch anything.

steam -cef-disable-gpu has no effect

Clean install of steam has no effect

Rolled back to Steam version 1705108172 and everything works as expected.

<details><summary>Latest console output:</summary>

```
rm: cannot remove '/home/dustin/.steam/steam': Is a directory
rm: cannot remove '/home/dustin/.steam/bin': Is a directory
steam.sh[1543]: Running Steam on devuan 5 64-bit
steam.sh[1543]: STEAM_RUNTIME is enabled automatically
setup.sh[1610]: Steam runtime environment up-to-date!
steam.sh[1543]: Steam client's requirements are satisfied
tid(1687) burning pthread_key_t == 0 so we never use it
[2024-03-02 15:22:15] Startup - updater built Feb 29 2024 00:39:10
[2024-03-02 15:22:15] Startup - Steam Client launched with: '/home/dustin/.steam/ubuntu12_32/steam' '-nominidumps' '-nobreakpad'
03/02 15:22:15 Init: Installing breakpad exception handler for appid(steam)/version(1709168962)/tid(1687)
[2024-03-02 15:22:15] Loading cached metrics from disk (/home/dustin/.steam/package/steam_client_metrics.bin)
[2024-03-02 15:22:15] Using the following download hosts for Public, Realm steamglobal
[2024-03-02 15:22:15] 1. https://client-update.akamai.steamstatic.com, /, Realm 'steamglobal', weight was 1000, source = 'update_hosts_cached.vdf'
[2024-03-02 15:22:15] 2. https://cdn.cloudflare.steamstatic.com, /client/, Realm 'steamglobal', weight was 1, source = 'update_hosts_cached.vdf'
[2024-03-02 15:22:15] 3. https://cdn.steamstatic.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
[2024-03-02 15:22:15] Verifying installation...
[2024-03-02 15:22:15] Verification complete
UpdateUI: skip show logo
Steam logging initialized: directory: /home/dustin/.steam/steam/logs

XRRGetOutputInfo Workaround: initialized with override: 0 real: 0xec683780
XRRGetCrtcInfo Workaround: initialized with override: 0 real: 0xec682050
steamwebhelper.sh[1716]: === Sat Mar  2 03:22:17 PM EST 2024 ===
steamwebhelper.sh[1716]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
Steam Runtime Launch Service: starting steam-runtime-launcher-service
Steam Runtime Launch Service: steam-runtime-launcher-service is running pid 1795
bus_name=com.steampowered.PressureVessel.LaunchAlongsideSteam
CAppInfoCacheReadFromDiskThread took 175 milliseconds to initialize
steamwebhelper.sh[1986]: === Sat Mar  2 03:22:27 PM EST 2024 ===
steamwebhelper.sh[1986]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
steamwebhelper.sh[2151]: === Sat Mar  2 03:22:37 PM EST 2024 ===
steamwebhelper.sh[2151]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
src/steamUI/steamuisharedjscontroller.cpp (546) : Failed creating offscreen shared JS context
src/steamUI/steamuisharedjscontroller.cpp (546) : Failed creating offscreen shared JS context
03/02 15:22:38 Init: Installing breakpad exception handler for appid(steam)/version(1709168962)/tid(1687)
assert_20240302152238_29.dmp[2323]: Uploading dump (out-of-process)
/tmp/dumps/assert_20240302152238_29.dmp
assert_20240302152238_29.dmp[2323]: Finished uploading minidump (out-of-process): success = yes
assert_20240302152238_29.dmp[2323]: response: CrashID=bp-9c6dc630-bd7c-42c9-bb13-e52472240302
assert_20240302152238_29.dmp[2323]: file ''/tmp/dumps/assert_20240302152238_29.dmp'', upload yes: ''CrashID=bp-9c6dc630-bd7c-42c9-bb13-e52472240302''
steamwebhelper.sh[2330]: === Sat Mar  2 03:22:47 PM EST 2024 ===
steamwebhelper.sh[2330]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
/usr/share/themes/Adwaita/gtk-2.0/main.rc:733: error: unexpected identifier 'direction', expected character '}'
/usr/share/themes/Adwaita/gtk-2.0/hacks.rc:28: error: invalid string constant "normal_entry", expected valid string constant
BRefreshApplicationsInLibrary 1: 1ms
steamwebhelper.sh[2532]: === Sat Mar  2 03:22:57 PM EST 2024 ===
steamwebhelper.sh[2532]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
Fossilize INFO: Autogroup scheduling is not enabled on this kernel. Will rely entirely on nice().
steamwebhelper.sh[2784]: === Sat Mar  2 03:23:08 PM EST 2024 ===
steamwebhelper.sh[2784]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
src/steamUI/steamuisharedjscontroller.cpp (546) : Failed creating offscreen shared JS context
src/steamUI/steamuisharedjscontroller.cpp (546) : Failed creating offscreen shared JS context
reaping pid: 2330 -- unknown
src/steamexe/main.cpp (264) : Assertion Failed: ReapProcess: waitid failed: 'No child processes'. Possibly leaking a zombie.

src/steamexe/main.cpp (264) : Assertion Failed: ReapProcess: waitid failed: 'No child processes'. Possibly leaking a zombie.

03/02 15:23:10 Init: Installing breakpad exception handler for appid(steam)/version(1709168962)/tid(1687)
assert_20240302152310_51.dmp[2969]: Uploading dump (out-of-process)
/tmp/dumps/assert_20240302152310_51.dmp
assert_20240302152310_51.dmp[2969]: Finished uploading minidump (out-of-process): success = yes
assert_20240302152310_51.dmp[2969]: response: CrashID=bp-1560df8a-793b-4d9b-8eff-9e8522240302
assert_20240302152310_51.dmp[2969]: file ''/tmp/dumps/assert_20240302152310_51.dmp'', upload yes: ''CrashID=bp-1560df8a-793b-4d9b-8eff-9e8522240302''
steamwebhelper.sh[2978]: === Sat Mar  2 03:23:18 PM EST 2024 ===
steamwebhelper.sh[2978]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
[2024-03-02 15:23:24] Shutdown
steam.sh[1543]: Restarting Steam by request...
rm: cannot remove '/home/dustin/.steam/steam': Is a directory
rm: cannot remove '/home/dustin/.steam/bin': Is a directory
steam.sh[1543]: Running Steam on devuan 5 64-bit
steam.sh[1543]: STEAM_RUNTIME has been set by the user to: /home/dustin/.steam/ubuntu12_32/steam-runtime
setup.sh[3291]: Steam runtime environment up-to-date!
steam.sh[1543]: Steam client's requirements are satisfied
tid(3409) burning pthread_key_t == 0 so we never use it
[2024-03-02 15:23:25] Startup - updater built Feb 29 2024 00:39:10
[2024-03-02 15:23:25] Startup - Steam Client launched with: '/home/dustin/.steam/ubuntu12_32/steam' '-nominidumps' '-nobreakpad'
03/02 15:23:25 Init: Installing breakpad exception handler for appid(steam)/version(1709168962)/tid(3409)
[2024-03-02 15:23:26] Loading cached metrics from disk (/home/dustin/.steam/package/steam_client_metrics.bin)
[2024-03-02 15:23:26] Using the following download hosts for Public, Realm steamglobal
[2024-03-02 15:23:26] 1. https://client-update.akamai.steamstatic.com, /, Realm 'steamglobal', weight was 1000, source = 'update_hosts_cached.vdf'
[2024-03-02 15:23:26] 2. https://cdn.cloudflare.steamstatic.com, /client/, Realm 'steamglobal', weight was 1, source = 'update_hosts_cached.vdf'
[2024-03-02 15:23:26] 3. https://cdn.steamstatic.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
[2024-03-02 15:23:26] Verifying installation...
[2024-03-02 15:23:26] Verification complete
UpdateUI: skip show logo
Steam logging initialized: directory: /home/dustin/.steam/steam/logs

XRRGetOutputInfo Workaround: initialized with override: 0 real: 0xec66e780
XRRGetCrtcInfo Workaround: initialized with override: 0 real: 0xec66d050
steamwebhelper.sh[3456]: === Sat Mar  2 03:23:27 PM EST 2024 ===
steamwebhelper.sh[3456]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
Steam Runtime Launch Service: starting steam-runtime-launcher-service
Steam Runtime Launch Service: steam-runtime-launcher-service is running pid 3623
bus_name=com.steampowered.PressureVessel.LaunchAlongsideSteam
CAppInfoCacheReadFromDiskThread took 181 milliseconds to initialize
steamwebhelper.sh[3818]: === Sat Mar  2 03:23:37 PM EST 2024 ===
steamwebhelper.sh[3818]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
steamwebhelper.sh[4040]: === Sat Mar  2 03:23:48 PM EST 2024 ===
steamwebhelper.sh[4040]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
src/steamUI/steamuisharedjscontroller.cpp (546) : Failed creating offscreen shared JS context
src/steamUI/steamuisharedjscontroller.cpp (546) : Failed creating offscreen shared JS context
03/02 15:23:48 Init: Installing breakpad exception handler for appid(steam)/version(1709168962)/tid(3409)
assert_20240302152348_29.dmp[4205]: Uploading dump (out-of-process)
/tmp/dumps/assert_20240302152348_29.dmp
assert_20240302152348_29.dmp[4205]: Finished uploading minidump (out-of-process): success = yes
assert_20240302152348_29.dmp[4205]: response: CrashID=bp-c79b86de-f118-4dfc-a1d7-b8f202240302
assert_20240302152348_29.dmp[4205]: file ''/tmp/dumps/assert_20240302152348_29.dmp'', upload yes: ''CrashID=bp-c79b86de-f118-4dfc-a1d7-b8f202240302''
steamwebhelper.sh[4213]: === Sat Mar  2 03:23:58 PM EST 2024 ===
steamwebhelper.sh[4213]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
/usr/share/themes/Adwaita/gtk-2.0/main.rc:733: error: unexpected identifier 'direction', expected character '}'
/usr/share/themes/Adwaita/gtk-2.0/hacks.rc:28: error: invalid string constant "normal_entry", expected valid string constant
BRefreshApplicationsInLibrary 1: 1ms
Fossilize INFO: Autogroup scheduling is not enabled on this kernel. Will rely entirely on nice().
Fossilize INFO: Autogroup scheduling is not enabled on this kernel. Will rely entirely on nice().
Fossilize INFO: Autogroup scheduling is not enabled on this kernel. Will rely entirely on nice().
steamwebhelper.sh[4586]: === Sat Mar  2 03:24:08 PM EST 2024 ===
steamwebhelper.sh[4586]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
steamwebhelper.sh[4748]: === Sat Mar  2 03:24:18 PM EST 2024 ===
steamwebhelper.sh[4748]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
src/steamUI/steamuisharedjscontroller.cpp (546) : Failed creating offscreen shared JS context
src/steamUI/steamuisharedjscontroller.cpp (546) : Failed creating offscreen shared JS context
reaping pid: 4213 -- unknown
src/steamexe/main.cpp (264) : Assertion Failed: ReapProcess: waitid failed: 'No child processes'. Possibly leaking a zombie.

src/steamexe/main.cpp (264) : Assertion Failed: ReapProcess: waitid failed: 'No child processes'. Possibly leaking a zombie.

03/02 15:24:21 Init: Installing breakpad exception handler for appid(steam)/version(1709168962)/tid(3409)
assert_20240302152421_52.dmp[4926]: Uploading dump (out-of-process)
/tmp/dumps/assert_20240302152421_52.dmp
assert_20240302152421_52.dmp[4926]: Finished uploading minidump (out-of-process): success = yes
assert_20240302152421_52.dmp[4926]: response: CrashID=bp-8f9697b1-67ee-42e2-a45d-697232240302
assert_20240302152421_52.dmp[4926]: file ''/tmp/dumps/assert_20240302152421_52.dmp'', upload yes: ''CrashID=bp-8f9697b1-67ee-42e2-a45d-697232240302''
^Csteamwebhelper.sh[4933]: === Sat Mar  2 03:24:28 PM EST 2024 ===
steamwebhelper.sh[4933]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
Steam Runtime Launch Service: steam-runtime-launcher-service pid 3623 exited
Steam Runtime Launch Service: starting steam-runtime-launcher-service
Steam Runtime Launch Service: steam-runtime-launcher-service is running pid 5095
bus_name=com.steampowered.PressureVessel.LaunchAlongsideSteam
[2024-03-02 15:24:39] Shutdown
```

</details>

<details><summary>Steamwebhelper log:</summary>
 
```
steamwebhelper.sh[321]: === Sat Mar  2 03:08:07 PM EST 2024 ===
steamwebhelper.sh[321]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/dustin/.steam/ubuntu12_64/steam-runtime-sniper
pressure-vessel-wrap[341]: W: "opt/amdgpu/share/libdrm" is unlikely to appear in "/run/host"
pressure-vessel-wrap[341]: W: "opt/amdgpu/share/libdrm" is unlikely to appear in "/run/host"
pressure-vessel-wrap[341]: W: "opt/amdgpu/share/drirc.d" is unlikely to appear in "/run/host"
pressure-vessel-wrap[341]: W: Disabling gtk3-nocsd LD_PRELOAD: it is known to cause crashes.
exec ./steamwebhelper --no-sandbox -lang=en_US -cachedir=/home/dustin/.steam/steam/config/htmlcache -steampid=28169 -buildid=1709168962 -steamid=76561197974114375 -logdir=/home/dustin/.
steam/steam/logs -uimode=7 -startcount=26 -steamuniverse=Public -realm=Global -clientui=/home/dustin/.steam/clientui -steampath=/home/dustin/.steam/ubuntu12_32/steam -launcher=0 -no-res
tart-on-ui-mode-change --enable-smooth-scrolling --password-store=basic --log-file=/home/dustin/.steam/steam/logs/cef_log.txt --disable-quick-menu --disable-features=DcheckIsFatal
./steamwebhelper: error while loading shared libraries: libdrm.so.2: cannot open shared object file: No such file or directory
```

</details>

#### Steps for reproducing this issue:

1. Run Steam
2. 
3.

**Comment by kisak-valve** [comment]
Hello @longshanks197, I agree that the vendored AMD driver in /opt is most likely a major contributing factor to the issue you're seeing.

Please completely close Steam, then run steam with `STEAM_LINUX_RUNTIME_VERBOSE=1 steam` from a terminal. This should add a significant amount of information to steamwebhelper.log for a Steam Runtime dev to ponder. Please attach the verbose log along with doing a general search for where libdrm.so.2 is located on your system. We're only really interested in where the library is supposed to be picked up from in this driver configuration and that it exists.

**Comment by longshanks197** [comment]
Well...I am a fool. At some point I symlinked /opt to /usr/opt as a quick fix for hard drive space instead of taking the time to resize the partition. I have resized the drive and moved the files into a proper /opt folder and everything is good. This is essentially the same as other issues with symlinks in essential steam search paths ([Like 10552](https://github.com/ValveSoftware/steam-for-linux/issues/10552)). Closing this issue to actions being taken.
